src/blockdata/opcodes.rs

```diff
@@ -803,11 +817,19 @@ macro_rules! ordinary_opcode {
     ($($op:ident),*) => (
         #[repr(u8)]
         #[doc(hidden)]
-        #[derive(Copy, Clone, PartialEq, Eq, Debug)]
+        #[derive(Copy, Clone, PartialEq, Eq)]
         pub enum Ordinary {
             $( $op = all::$op.code ),*
         }
 
+        impl fmt::Display for Ordinary {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                match *self {
+                    $(Ordinary::$op => {f.pad(&format!("{}", all::$op))}),*
```

Comment:
This allocates which could be unfortunate for long scripts. Thankfully there is an easy solution:

```rust
$(Ordinary::$op => { f.pad(stringify!($op)) }),*
```

should work unless I'm missing somethin.